pyccel/parser/semantic.py

```diff
@@ -1459,6 +1459,12 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign, *
                             bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
                             severity='fatal', blocker=False)
 
+                elif var.is_ndarray and var.is_pointer:
+                        # we allow pointers to be reassigned multiple times
+                        # pointers reassigning need to call free_pointer func
+                        # to remove memory leaks
+                        new_expressions.append(Deallocate(var))
```

Comment:
why did you move this condition out of if block?
```
elif d_var['shape'] != shape:   # line 1504
```

Reply:
I don't see this modification

Reply:
the reassign of a view need to be deallocated even if the shapes are the same
they can be view of different arrays
so I needed to make the check outside the if block of that checks the shape